provision/provision.py

```diff
@@ -0,0 +1,157 @@
+# -*- coding: utf-8 -*-
+"""
+$ provision/provision.py {setup,teardown,populate,raze} <domain>
+"""
+import json
+import logging
+import os.path
+import string
+import sys
+import time
+import uuid
+
+import boto3
+
+logging.basicConfig(level=logging.INFO)
+logger = logging.getLogger(__name__)
+es = boto3.client('es')
+
+
+def getpath(filename):
+    return os.path.abspath(os.path.join(os.path.dirname(__file__), filename))
+
+
+def get_legacy_client(endpoint):
+    sys.path.insert(0, getpath('..'))
+    os.environ['ES_HOST'] = endpoint[7:]  # remove http://
+    from app import client
+    return client
+
+
+def get_endpoint(es_domain):
+    """
+    Finds the endpoint (that requests should be made to) for a given
+    ElasticSearch domain.
+
+    :rtype: string
+    """
+    backoff = 0
+    status = {}
+    while not status.get('DomainStatus', {}).get('Endpoint', None):
+        status = es.describe_elasticsearch_domain(DomainName=es_domain)
+        # \sum_{n=0}^{9} 2^n is a long time and if there hasn't been a response
+        # by then, one probably isn't coming anytime soon.
+        if backoff > 10:
+            raise RuntimeError("It's been a long time, and the ElasticSearch "
+                               "domain still looks to be unavailable. It's "
+                               "been long enough that I'm considering this "
+                               "a timeout.")
+        if not status['DomainStatus'].get('Endpoint', None):
+            time.sleep(2 ** backoff)
+            backoff += 1
+
+    # boto3 returns the endpoint without the URL protocol, so we need
+    # to add it if it doesn't exist, otherwise requests will choke
+    return 'http://' + status['DomainStatus']['Endpoint']
+
+
+def populate_domain(endpoint):
+    """
+    Given an ElasticSearch endpoint, populates the domain with a
+    'faithful' replica of dss-azul-commons for testing purposes.
+    """
+    # boto3 can't sign requests by default, so we fall back to boto2
+    client = get_legacy_client(endpoint)
+
+    # Set up fb_index (data_objects)
+    with open(getpath('index-mapping.json'), 'r') as data:
+        payload = json.load(data)['fb_index']
+    client.make_request(method='PUT', path='/fb_index', data=json.dumps(payload))
```

Comment:
Since `fb_index` is configurable in the config.json, one could end up configuring such that we're not loading the index we're checking. Should this pull from the config too?

Not blocking, but worth mentioning.